### PR TITLE
Moves the multi-processing stuff out of main functions

### LINKED ISSUE #91: Add documentation for config params

### PR DESCRIPTION
**Pull request by Sean1572**
https://github.com/pytorch/pytorch/issues/3492

Seems for sweeps we need to call `torch.multiprocessing.set_start_method('spawn')` once.  I moved this function outside of train_main() and called it only once for sweeps. Hoping this fixes the issue, currently running this on current sweeps to test the restart function.

### CODE REVIEW

**Comment by Sean1572** [comment]
image
There error in question

File: classification/config.py [hunk 1]
```diff
@@ -32,6 +30,7 @@ def __init__(self):
         self.get_git_hash()
         self.cli_values()
         self.get_device()
+        self.device = "cpu"
```
Comment (benjamin-cates) [inline classification/config.py]: ermmmm should prolly remove this

File: documentation/default_config.yml [hunk 2]
```diff
@@ -9,12 +9,17 @@ offset_col: "OFFSET"
 duration_col: "DURATION"
 file_name_col: "FILE NAME"
 manual_id_col: "MANUAL ID"
-
+cpu_preprocessing: True
```
Comment (benjamin-cates) [inline documentation/default_config.yml]: why is this coming back?

File: classification/config.py (hunk 1 above)
Comment (marco-bazzani) [reply classification/config.py]: yup

**Comment by marco-bazzani** [comment]
Still getting some errors, not ready to merge

Specifically: TypeError("unhashable type: 'Series'")

I'm guessing that at some point a Series is getting stored as a key for the config_dict?

**Comment by marco-bazzani** [comment]
Merging may have fixed the issue? No crashes in 10/10 runs so far. Would appreciate if someone else could test as well to double check this?

**Comment by marco-bazzani** [comment]
What's up guys? It's me, your friendly neighborhood dev here. I just wanted to let you know that I've totally tubularly tested this branch and gotten it ready to merge. It'd be swag if any of my brothers and sisters out there could give it a run and tell me about any groovy error messages they see ✌️✌️☮️☮️🌷🌷

**Review by benjamin-cates — APPROVED** [review]
Looks pretty slay 💅 Works on my machine :3     <3 and tests pass

**Review by sprestrelski — APPROVED** [review]
TM, LG

**Pull request merged**